### PR TITLE
Handle negative height & width in image annotations

### LINKED ISSUE #7083: It is possible to create image annotations that cannot be selected
#### Summary

It is possible to create unselectable (and thus unremovable) image annotations.

#### Expected vs Current Behavior

Image annotations should always be selectable.

#### Steps to Reproduce

Reproduction steps semi-unknown at this point.

1. Use CouchDB persistence or test in our testing environment
2. Navigate to a display layout with real imagery
3. Hold option+shift and click and drag from the BOTTOM RIGHT to the TOP LEFT to create the annotation box
4. Tag it
5. The annotation is unselectable and unremovable.

#### Environment

* Open MCT Version:  3.1.0-next
* Deployment Type: 
* OS:
* Browser:

#### Impact Check List

- [x] Data loss or misrepresented data?
- [x] Regression? Did this used to work or has it always been broken?
- [ ] Is there a workaround available?
- [x] Does this impact a critical component?
- [ ] Is this just a visual bug with no functional impact?
- [ ] Does this block the execution of e2e tests?
- [ ] Does this have an impact on Performance?

#### Additional Information

https://github.com/nasa/openmct/assets/9259993/a9805e3a-979c-4471-b4f9-845ced7e0930

### PR DESCRIPTION
**Pull request by scottbell**
Closes #7083 

### Describe your changes:

If the height or width are negative due the user dragging back and to the left, or forward and up, handle this properly when adding to the flatbush index.

### All Submissions:

* [x] Have you followed the guidelines in our [Contributing document](https://github.com/nasa/openmct/blob/master/CONTRIBUTING.md)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/nasa/openmct/pulls) for the same update/change?
* [x] Is this change backwards compatible? For example, developers won't need to change how they are calling the API or how they've extended core plugins such as Tables or Plots.

### Author Checklist

* [x] Changes address original issue?
* [x] Tests included and/or updated with changes?
* [x] Command line build passes?
* [x] Has this been smoke tested?
* [x] Testing instructions included in associated issue OR is this a dependency/testcase change?

### Reviewer Checklist

* [ ] Changes appear to address issue?
* [ ] Reviewer has tested changes by following the provided instructions?
* [ ] Changes appear not to be breaking changes?
* [ ] Appropriate automated tests included?
* [ ] Code style and in-line documentation are appropriate?
* [ ] Has associated issue been labelled unverified? (only applicable if this PR closes the issue)
* [ ] Has associated issue been labelled bug? (only applicable if this PR is for a bug fix)

### CODE REVIEW

File: e2e/tests/functional/plugins/imagery/exampleImagery.e2e.spec.js
Comment (scottbell) [inline e2e/tests/functional/plugins/imagery/exampleImagery.e2e.spec.js:1]: Hilariously, we already creating wacko rectangles in this test, we just didn't try to click on them afterwards!

File: src/plugins/imagery/components/AnnotationsCanvas.vue [hunk 1]
```diff
@@ -78,12 +78,21 @@ export default {
...
+
+          // if height or width are negative, we need to adjust the x and y
+          if (width < 0) {
+            x2 = x;
+            x = x + width;
+          }
+          if (height < 0) {
+            y2 = y;
+            y = y + height;
+          }
```
Comment (ozyx) [inline src/plugins/imagery/components/AnnotationsCanvas.vue:93]: Should we extract this into its own function? Might make this bit more readable

File: e2e/tests/functional/plugins/imagery/exampleImagery.e2e.spec.js [hunk 2]
```diff
@@ -222,7 +221,7 @@ test.describe('Example Imagery Object', () => {
     await expect(page.locator('[role="toolbar"][aria-label="Image controls"]')).toBeVisible();
     await Promise.all(tagHotkey.map((x) => page.keyboard.up(x)));
 
-    //Wait for canvas to stabilize.
+    // Wait for canvas to stabilize.
```
Comment (ozyx) [inline e2e/tests/functional/plugins/imagery/exampleImagery.e2e.spec.js:224]: let's add an annotation to this test referencing the issue

**Review by ozyx — CHANGES_REQUESTED** [review]
Awesome! Just a couple minor suggestions before we merge

File: src/plugins/imagery/components/AnnotationsCanvas.vue [hunk 3]
```diff
@@ -124,6 +122,23 @@ export default {
       this.selectedAnnotations = annotations;
       this.$emit('annotations-changed', annotations);
     },
+    transformAnnotationRectangleToFlatbushRectangle(annotationRectangle) {
```
Comment (ozyx) [inline src/plugins/imagery/components/AnnotationsCanvas.vue:125]: perfect!

**Review by ozyx — APPROVED** [review]
🤌

**Pull request merged**